sdk/confidentialledger/azure-data-confidentialledger/README.md

```diff
@@ -0,0 +1,155 @@
+```xml
+<dependency>
+  <groupId>com.azure</groupId>
+  <artifactId>azure-data-confidentialledger</artifactId>
+  <version>1.0.0-beta.1</version>
+</dependency>
+```
+[//]: # ({x-version-update-end})
+
+### Authenticate the client
+
+#### Using Azure Active Directory
+
+In order to interact with the Azure Confidential Ledger service, your client must present an Azure Active Directory bearer token to the service.
+
+The simplest way of providing a bearer token is to use the `DefaultAzureCredential` authentication method by providing client secret credentials is being used in this getting started section but you can find more ways to authenticate with [azure-identity][azure_identity].
+
+#### Using a client certificate
+
+As an alternative to Azure Active Directory, clients may choose to use a client certificate to authenticate via mutual TLS. `CertificateCredential` may be used for this purpose.
+
+#### Create LedgerBaseClient with Azure Active Directory Credential
+
+You can authenticate with Azure Active Directory using the [Azure Identity library][azure_identity].
+
+To use the [DefaultAzureCredential][DefaultAzureCredential] provider shown below, or other credential providers provided with the Azure SDK, please include the `azure-identity` package:
+
+[//]: # ({x-version-update-start;com.azure:azure-identity;dependency})
+```xml
+<dependency>
+    <groupId>com.azure</groupId>
+    <artifactId>azure-identity</artifactId>
+    <version>1.2.5</version>
+</dependency>
+```
+
+Set the values of the client ID, tenant ID, and client secret of the AAD application as environment variables: AZURE_CLIENT_ID, AZURE_TENANT_ID, AZURE_CLIENT_SECRET.
+
+##### Example
+<!-- embedme ./src/samples/java/com/azure/data/confidentialledger/ReadmeSamples.java#L32-L56 -->
+```java
+ConfidentialLedgerIdentityServiceBaseClient identityServiceClient = new ConfidentialLedgerClientBuilder()
+    .ledgerUri(new URL("<confidential-ledger-url>"))
+    .identityServiceUri(new URL("<confidential-ledger-identity-service-url>"))
+    .credential(new DefaultAzureCredentialBuilder().build())
+    .buildConfidentialLedgerIdentityServiceBaseClient();
+
+String ledgerId = "<confidential-ledger-url>"
+    .replaceAll("\\w+://", "")
+    .replaceAll("\\..*", "");
+DynamicResponse response = identityServiceClient.getLedgerIdentity(ledgerId).send();
+JsonReader jsonReader = Json.createReader(new StringReader(response.getBody().toString()));
+JsonObject result = jsonReader.readObject();
+String tlsCert = result.getString("ledgerTlsCertificate");
+reactor.netty.http.client.HttpClient reactorClient = reactor.netty.http.client.HttpClient.create()
+    .secure(sslContextSpec -> sslContextSpec.sslContext(SslContextBuilder.forClient()
+        .trustManager(new ByteArrayInputStream(tlsCert.getBytes(StandardCharsets.UTF_8)))));
+HttpClient httpClient = new NettyAsyncHttpClientBuilder(reactorClient).wiretap(true).build();
```

Comment:
This should be supported in `HttpClientOptions` - @alzimmermsft

Reply:
Is this directed at the SSL context or wiretap?